core/src/androidMain/kotlin/com/ustadmobile/core/domain/share/SendAppFileUseCaseAndroid.kt

```diff
@@ -0,0 +1,40 @@
+package com.ustadmobile.core.domain.share
+
+import android.content.Context
+import android.content.Intent
+import androidx.core.content.FileProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+
+
+class SendAppFileUseCaseAndroid(
+    private val appContext: Context
+) : SendAppFileUseCase {
+    override suspend operator fun invoke() {
+        withContext(Dispatchers.IO) {
+            val apkFile = File(appContext.applicationInfo.sourceDir)
+            val apkUri = FileProvider.getUriForFile(
+                appContext,
+                "${appContext.packageName}.provider",
+                apkFile
+            )
+
+            // Split name check
+            val splitNames = appContext.packageManager.getPackageInfo(appContext.packageName, 0).splitNames
+            if (splitNames != null && splitNames.isNotEmpty()) {
+                throw IllegalArgumentException("APK has splits and cannot be shared as a single file.")
+            }
+
+            val intent = Intent(Intent.ACTION_SEND).apply {
+                type = "application/vnd.android.package-archive"
+                putExtra(Intent.EXTRA_STREAM, apkUri)
+                addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+            }
+
+            val chooserIntent = Intent.createChooser(intent, "Share App")
```

Comment:
"Share App" - this needs to be added to the string xml. 

Catch: AppContext MUST be the activity, not the application context (otherwise localization will be wrong). Bind the use case in the activity, not in UstadApp.